core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -3670,7 +3659,6 @@ private void transitiveInference(RelOptRule... extraRules) throws Exception {
     transitiveInference();
   }
 
-  @Ignore("not working")
   @Test public void testTransitiveInferenceUnion3wayOr() throws Exception {
```

Comment:
These tests looks weird, they have the same test contents but split to multiple methods, can we just merge them into one ?

Reply:
hi, @danny0405 , thanks for reviewing.
To my understanding, these tests have same work contents, but they have different sql, before plan  and after plan value in `RelOptRules.xml`. So, might just keep them separate as it is?

Reply:
Yeah, these tests fetch the test sql from the xml file which is different from the other plan tests, i kind of think this way is not that intuitive, we can only see the test sql in the xml file. Can we do some refactoring for this method to let each test method pass in their own test sql ?

Reply:
OK, I got your point.
I update the PR, and add the sql as input parameter for `transitiveInference` method.
I add a check to make sure the sql parameter passed is the same with sql in `RelOptRules.xml`.